src/librustc_trans/lib.rs

```diff
@@ -180,3 +180,55 @@ pub struct CrateTranslation {
 }
 
 __build_diagnostic_array! { librustc_trans, DIAGNOSTICS }
+
+use rustc::session::Session;
+pub fn init(sess: &Session) {
```

Comment:
Could you make this return a token so that you can't call the 10 or so public trans functions without calling `init` (AFAICT half of these - `provide`, `find_crate_name`, `filename_for_input`, `default_output_for_target`, `invalid_output_for_target` - don't need LLVM, but a few others do)?